### DISCUSSION TITLE
What exactly is the default post.excerpt?

### DISCUSSION THREAD

**Discussion by Jmuccigr** [jekyll/jekyll#9685] [Q&A]
I see in various places that the default post.excerpt is the first paragraph, but that it can be reset by defining an excerpt_separator which will then be used to set the end of the excerpt. However in my feed the default excerpt—that is, in the absence of an excerpt_separator—is equivalent to the content. If there's an excerpt_separator in there, it gets respected.

Am I missing something or is this a bug?

**Comment by ashmaroli** [comment]
The default *excerpt* is the block of content string **before two consecutive Unix-style line-breaks (`\n\n`)**.

So, if the content string of a post was:
```
The quick brown fox

jumped over the sleeping lazy dog.
```
The content string is:
```
"The quick brown fox\n\njumped over the sleeping lazy dog."
```
Therefore, the default excerpt would be:
```
The quick brown fox
```

**Comment by Jmuccigr** [comment]
Thanks. That's a more specific answer than the basic Jekyll pages have. :-)

I've played around a bit more and noticed that when there's an `excerpt_separator` defined in _config.yml, jekyll will gobble up the entire contents as the excerpt if no separator is present. I was expecting it to apply the default in the absence of a separator, but the definition of a separator eliminates the default.

**Comment by Jmuccigr** [comment]
Following up on my last comment, I wonder how a request to change this behavior would be received. 

To my mind the excerpt_separator should serve to indicate behavior different from the default when the user wants that. If it’s not present, the default separator should be applied. This allows for the default behavior except when the user wants something different. 

Most of my use cases are when I’ve intentionally put in text I want to serve as a summary. In other cases I’m happy to have the default behavior take over.

**Comment by ashmaroli** [comment]
I think you are still not clear on the concept as a whole.
Let me try and shed some light using the following sample content:

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.

<!-- excerpt_end -->

Sed risus diam, efficitur eget quam et, aliquet dictum quam.
```

## Default Behavior:
### Excerpt:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

### Content:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.



Sed risus diam, efficitur eget quam et, aliquet dictum quam.

## Custom Behaviors
### When configured to Not Generate an Excerpt (`excerpt_separator: ""`)

#### Excerpt:

#### Content:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.



Sed risus diam, efficitur eget quam et, aliquet dictum quam.

### When configured with a custom *Excerpt Separator* (`excerpt_separator: "<!-- excerpt_end -->"`)

#### Excerpt:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.


#### Content:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.



Sed risus diam, efficitur eget quam et, aliquet dictum quam.

### When configured with a custom *Excerpt* (`excerpt: "This is a dummy text.\n\nRead More &raquo;"` in document's *front matter*)

#### Excerpt:

This is a dummy text.

Read More &raquo;

#### Content:

Lorem ipsum dolor sit amet, consectetur adipiscing elit.

Suspendisse porta mattis nulla ac sollicitudin. Curabitur lobortis ac metus sit amet imperdiet. Aliquam id pulvinar lorem,
ac volutpat lorem. Ut vulputate sollicitudin eros, dapibus lobortis lectus ultrices a. Suspendisse et lacus vulputate orci
auctor laoreet.



Sed risus diam, efficitur eget quam et, aliquet dictum quam.

**Reply by Jmuccigr** [reply]
No, I get it. :-)

My use case is fairly simple: normally I'm happy with the default separator. However there are times when I want to specify an excerpt which doesn't work with the default. For that I can specify my own separator. However if I specify it in the config file, that works for the whole feed, not just for a specific post, so either I insert my separator into every post, or the excerpt is equivalent to the entire contents of the post.

What I'm suggesting is that it might be better for it to work like this: if the user specifies a separator and that separator isn't present in a post, use the default separator instead, rather than including all the content. That might be computationally more work, of course.

That said, I now see that I can specify the separator in the front matter which means I can include it only when I want to. I still like my suggestion, but I can live with the front-matter approach.

**Reply by ashmaroli** [reply]
> if the user specifies a separator and that separator isn't present in a post, use the default separator instead

Ah! You're suggesting to use the *default* `excerpt_separator` as a fallback mechanism. :)
This behavior-change is best suited for a future major-version-bump release.

You may open a feature-request ticket to register the idea. The future-me / active-maintainer at the time will not skim through all discussions.

**Reply by Jmuccigr** [reply]
Yes, that's it exactly. Thanks for the suggestion.
